src/reducers/WeatherReducer.js

```diff
@@ -0,0 +1,18 @@
+import { FETCH_WEATHER } from '../actions';
+
+const initialState = {
+  weather: [],
+};
+
+const WeatherReducer = function (state = initialState, action) {
+  switch (action.type) {
+    case FETCH_WEATHER:
+      return {
+        weather: [...state.weather, action.payload],
```

Comment:
No need to reference `weather` here - you're returning this value to your store which already has a `weather` key